Source/Core/Common/WorkQueueThread.h

```diff
@@ -20,89 +21,148 @@ class WorkQueueThread
 {
 public:
   WorkQueueThread() = default;
-  WorkQueueThread(std::function<void(T)> function) { Reset(std::move(function)); }
+  WorkQueueThread(const std::string name, std::function<void(T)> function)
+  {
+    Reset(std::move(name), std::move(function));
+  }
   ~WorkQueueThread() { Shutdown(); }
-  void Reset(std::function<void(T)> function)
+
+  // Shuts the current work thread down (if any) and starts a new thread with the given function
+  // Note: Some consumers of this API push items to the queue before starting the thread.
+  void Reset(const std::string& name, std::function<void(T)> function)
   {
     Shutdown();
-    m_shutdown.Clear();
-    m_cancelled.Clear();
+    std::lock_guard lg(m_lock);
+    m_thread_name = std::move(name);
```

Comment:
It's generally a bad idea to move a `const` value.  I'd either pass name by value or possibly use a `string_view` (though that would create a copy here)